### PR TITLE
Automation component

### PR DESCRIPTION
**Pull request by marinmitev**
Automation API and JSON files - initial contribution by ProSyst

Author: Marin Mitev <marin.mitev@gmail.com>
Also-by: Jordan Mihailov <j.mihailov@prosyst.com>, Vasil Ilchev <v.ilchev@prosyst.bg>, Ana Dimova <a.dimova@prosyst.bg>

### CODE REVIEW

**Comment by kaikreuzer** [comment]
Marin, thanks a lot for this PR! We will make look at it asap and give you feedback!

**Comment by kaikreuzer** [comment]
Thanks for squashing, Marin!
Unfortunately, the CLA validation fails - did you use by mistake a wrong e-mail address for signing?
Please also make sure to list any colleagues who were involved as "Also-by", see https://wiki.eclipse.org/Development_Resources/Contributing_via_Git#The_Commit_Record.

Next step is for me then to create a CQ, i.e. to pass the code on to the IP team at Eclipse. Thanks again!

**Comment by marinmitev** [comment]
Hi Kai, it seems I have not completed the CLA process, and I have just did it
Please check again
Thanks,
Marin

**Comment by kaikreuzer** [comment]
Thanks Marin,

but unfortunately, there are still two issues:
1. There are now 3 commits instead of 1 - you will have to squash it again
2. All "Also-by" contributors need to have a signed CLA - this does not seem to be the case.

Cheers,
Kai

**Comment by dnobel** [comment]
Moreover the "Signed-off-by <author>" line is missing. The format must be:
```
Some commit message

Also-by: Some Bodyelse1 <somebodyelse1@nowhere.com>
Also-by: Some Bodyelse2 <somebodyelse2@nowhere.com>
Signed-off-by: Some Body <somebody@somewhere.com>
```

**Comment by marinmitev** [comment]
Hi again,
squashed all commits to one;
splitted the co-authors in separate lines;
added signed off account;
and created eclipse accounts with signed CLA agreement
Now I hope everything is OK,
Best regards,
Marin

**Comment by dnobel** [comment]
Thanks Marin! As the message "All is well — All committers passed Eclipse CLA and Signed-off-by validation." indicates, now everything is fine (I also rechecked the CLA for every author manually). So we can merge the PR.

**Comment by kaikreuzer** [comment]
wonderful, thanks!
But @dnobel, we cannot merge yet, I now first have to create a CQ for this PR!

**Comment by dnobel** [comment]
Sorry, you are right. Definitely more than 1000 lines

File: bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/template/TemplateProvider.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/template/TemplateProvider.java]: Could you fix the license header to be EPL?

File: bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/type/ModuleTypeProvider.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/type/ModuleTypeProvider.java]: Could you fix the license header to be EPL?

File: bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Connection.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2015 by ProSyst Software GmbH
+ * http://www.prosyst.com
+ * All rights reserved.
+ *
+ * This software is the confidential and proprietary information
```
Comment (kaikreuzer) [inline bundles/automation/org.eclipse.smarthome.automation.api/src/main/java/org/eclipse/smarthome/automation/Connection.java]: Could you fix the license header to be EPL?

**Comment by kaikreuzer** [comment]
@marinmitev , when creating the CQ I stumbled across three license headers that are not correct - so I have to ask you for yet another fix and a squash!

**Comment by marinmitev** [comment]
Kai, sorry for that - we have overlooked these 3 files, now it should be ok

**Comment by kaikreuzer** [comment]
Thanks again! I have now created CQ9374, hence we will now have to wait for the approval of the IP team.

**Comment by kaikreuzer** [comment]
Good news everybody! We have just received the approval to merge this PR in accordance with [Parallel
IP](https://wiki.eclipse.org/Development_Resources/HOWTO/Parallel_IP_Process).

**Pull request merged**